Add query limiter middleware for instant queries (#11097)
grafana/mimir: CHANGELOG.md

* [ENHANCEMENT] Ingester: Remove cost-attribution experimental `max_cost_attribution_labels_per_user` limit. #11090
* [ENHANCEMENT] Update Go to 1.24.2. #11114
* [ENHANCEMENT] Ingester/Distributor: Add `cortex_cost_attribution_*` metrics to observe the state of the cost-attribution trackers. #11112

> * [ENHANCEMENT] Add configurable per-tenant `limited_queries`, which may only be executed at or less than a certain frequency. #11097

Reviewer: ```suggestion
* [ENHANCEMENT] Add configurable per-tenant `limited_queries`, which you can only run at or less than a certain frequency. #11097
```

* [ENHANCEMENT] Update Go to 1.24.2. #11114
* [ENHANCEMENT] Ingester/Distributor: Add `cortex_cost_attribution_*` metrics to observe the state of the cost-attribution trackers. #11112
* [ENHANCEMENT] gRPC/HTTP servers: Add `cortex_server_invalid_cluster_validation_label_requests_total` metric, that is increased for every request with an invalid cluster validation label. #11241

> * [ENHANCEMENT] Add configurable per-tenant `limited_queries`, which you can only run at or less than an allowed frequency. #11097

Author: We don't currently document this for any of the per-tenant config blocks under [`limits`](https://grafana.com/docs/mimir/latest/configure/configuration-parameters/#limits), so I think we'd need more input from docs (i.e., @tacole02) about how to represent this, or how much support makes sense given these are all experimental.

Here's an example `runtime.yaml` when running the local microservices deployment (using the example query in the tests):
```
overrides:
  "anonymous":
    limited_queries:
      - query: "rate(metric_counter[5m])"
        allowed_frequency: "1m"
```

> * [ENHANCEMENT] Add configurable per-tenant `limited_queries`, which you can only run at or less than an allowed frequency. #11097

Reviewer: I agree, I think we should add them both to https://grafana.com/docs/mimir/latest/configure/configuration-parameters/#limits and mark that they are experimental. If either of you want to draft up descriptions for these, I'm happy to take a look.

> * [ENHANCEMENT] Add configurable per-tenant `limited_queries`, which you can only run at or less than an allowed frequency. #11097

Author: I can add an example block for limited queries in this PR, but I'd like to scope it to just that and I'm happy to add example blocks for blocked requests and queries in a separate PR.

> * [ENHANCEMENT] Add configurable per-tenant `limited_queries`, which you can only run at or less than an allowed frequency. #11097

Author: > I can add an example block for limited queries in this PR

This is in.
